sw/device/silicon_creator/mask_rom/sigverify_keys.c

```diff
@@ -158,30 +184,77 @@ static rom_error_t dev_key_is_valid(lifecycle_state_t lc_state,
  */
 static rom_error_t key_is_valid(sigverify_key_type_t key_type,
                                 lifecycle_state_t lc_state, size_t key_index) {
-  switch (key_type) {
-    case kSigverifyKeyTypeTest:
-      return test_key_is_valid(lc_state, key_index);
-    case kSigverifyKeyTypeProd:
-      return prod_key_is_valid(lc_state, key_index);
-    case kSigverifyKeyTypeDev:
-      return dev_key_is_valid(lc_state, key_index);
+  switch (launder32(lc_state)) {
+    case kLcStateTestUnlocked0:
+      HARDENED_CHECK_EQ(lc_state, kLcStateTestUnlocked0);
```

Comment:
Would it be better to pass `lc_state` into the `key_is_valid_in_*` function and do this check there (or even better do the check again there)?